modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -61,12 +72,32 @@
     public Node getNode();
 
     /**
-     * Called by a Skinnable when the Skin is replaced on the Skinnable. This method
-     * allows a Skin to implement any logic necessary to clean up itself after
-     * the Skin is no longer needed. It may be used to release native resources.
+     * Called once when {@link Skin} is set, after the
+     * previous skin, if any, has been uninstalled via its {@link #dispose()} method.
+     * This method allows a Skin to register listeners, add child nodes, set
+     * required properties and/or event handlers.
+     * <p>
+     * Application code must not call this method.
+     * <p>
+     * The default implementation of this method does nothing.
+     *
+     * @implNote
+     * Most implementations of Skin
+     * do not need to implement {@code install} unless they must set one or more
+     * properties in the corresponding Skinnable.
+     *
+     * @since 20
+     */
+    default public void install() { }
+
+    /**
+     * Disconnects the {@link Skin} from its {@link Skinnable}.
+     * This method allows a {@link Skin} to implement any logic necessary to clean up itself after
+     * the {@code Skin} is no longer needed. It may be used to release native resources.
      * The methods {@link #getSkinnable()} and {@link #getNode()}
```

Comment:
I don't think this is correct "Disconnects the skin from its skinnable" -- the control does this.  I also think "clean up" covers the "native resources" bit, and not something that should be mentioned in an interface description.

May I suggest:

> Called when a previously installed skin is about to be removed from its associated control. This allows the skin to do clean up, like removing listeners and bindings, and undo any permanent changes to its control.  After this method completes, {@link #getSkinnable()} and {@link #getNode()} should return {@code null}.

Reply:
I like this too, except for "permanent", which I  think is a misnomer.